lib/mongoid/config.rb

```diff
@@ -216,6 +216,7 @@ def load_configuration(settings)
       configuration = settings.with_indifferent_access
       self.options = configuration[:options]
       self.clients = configuration[:clients]
+      Mongo.options = configuration[:driver_options] || {}
```

Comment:
One of these things is not like the other... here's why: https://github.com/mongodb/mongo-ruby-driver/pull/2515#discussion_r883942106

Reply:
I think this should iterate the contents of `configuration[:driver_options]` and call `Mongo.send("#{option}=", value)`.

Reply:
Which also would stop discarding options that are already set for driver and don't exist in Mongoid config.yml. In fact there should be a test for this.

Reply:
If an option value is nil here would you write nil to driver's option or do nothing?

Reply:
That's what the drivers `options=` method does: 
``` 
options.each_pair do |option, value|
  Validators::Option.validate(option)
  send("#{option}=", value)
end

Reply:
I have added that test

Reply:
> If an option value is nil here would you write nil to driver's option or do nothing?

If the option is unset, nothing, if the option is set to `nil`, set the driver's option

Reply:
Can you add a test for an option with nil value please.

Reply:
as there are only two options available in Mongo, I cannot. One is already being used for regular assignment, and one is being used to show that it does not get overridden.

Reply:
Add another config file?

Reply:
oh lol yeah good point

Reply:
done